src/main/java/mate/academy/web_intro_bookstore/service/BookServiceImpl.java

```diff
@@ -40,4 +41,39 @@ public BookDto findById(Long id) {
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, UpdateBookRequestDto updateDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityBotFoundException("Book not found with id: " + id));
+
+        if (updateDto.getTitle() != null) {
+            book.setTitle(updateDto.getTitle());
+        }
+
+        if (updateDto.getAuthor() != null) {
+            book.setAuthor(updateDto.getAuthor());
+        }
+
+        if (updateDto.getPrice() != null) {
+            book.setPrice(updateDto.getPrice());
+        }
+
+        if (updateDto.getDescription() != null) {
+            book.setDescription(updateDto.getDescription());
+        }
+
+        if (updateDto.getCoverImage() != null) {
+            book.setCoverImage(updateDto.getCoverImage());
+        }
+
+        Book savedBook = bookRepository.save(book);
+
+        return bookMapper.toDto(savedBook);
```

Comment:
create method in mapper to update book
example
`void updateBookFromDto(CreateBookRequestDto book, @MappingTarget Book entity);`
```suggestion
        bookMapper.updateBookFromDto(updateDto, book);
        bookRepository.save(book);
        return bookMapper.toDto(book);
```